book/src/debugging.md

```diff
@@ -0,0 +1,63 @@
+# Debugging
+
+Division gives us a perfect opportunity to exercise debugging skills. Write a
+test in `jnibookjava` that divides by zero, then run it. It's likely that the
+JVM will crash and you'll see a message similar to this:
+
+```
+``thread '<unnamed>' panicked at 'attempt to divide by zero', src/division.rs:14:5
+note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+fatal runtime error: failed to initiate panic, error 5
+```
+
+Dividing by zero triggers a panic. All panics across the FFI boundary cause
+undefined behavior. In this chapter, we'll learn about how to debug native 
+libraries.
+
+## Putting a Debugger in a Debugger
+
+### Java Instructions
+
+Run a test with debug and add a breakpoint here:
+
+```
+●   static native int divide(int a, int b);
+```
+
+You'll need to find the PID of your test. If you're using Java 11, then you can
+fetch it using the `ProcessHandle` API, as shown below. Otherwise, you'll need
+to find the PID through other means, such as through `ps -x` on Unix-like
```

Comment:
There is [`jps`](https://docs.oracle.com/javase/9/tools/jps.htm#JSWOR733) coming with any JDK, possibly, it is also available on Windows.

The docs say though "experimental and unsupported" :)